### PR TITLE
Tests for oneAPI bfloat16 extension

### PR DESCRIPTION
**Pull request by AmirIpma**
Created tests for [bfloat16](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_bfloat16.asciidoc) extension

### CODE REVIEW

File: tests/extension/oneapi_bfloat16/bfloat16_api.cpp [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    CHECK(std::is_convertible_v<bfloat16, float>);
+    CHECK(std::is_convertible_v<bfloat16, sycl::half>);
+    CHECK(std::is_convertible_v<bfloat16, bool>);
+    CHECK(std::is_convertible_v<float, bfloat16>);
+    CHECK(std::is_convertible_v<sycl::half, bfloat16>);
+  }
+}
+
+TEST_CASE("Special values", "[bfloat16]") {
+  SECTION("Check that bfloat16 occupies 16 bits of memomry") {
```
Comment (keryell) [inline tests/extension/oneapi_bfloat16/bfloat16_api.cpp:65]:
```suggestion
  SECTION("Check that bfloat16 occupies 16 bits of memory") {
```
Configure your spell-checker.

File: tests/extension/oneapi_bfloat16/bfloat16_api.cpp [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    REQUIRE(sizeof(bfloat16) == 2);
+  }
+
+  SECTION("Check minimum positive normal value") {
+    uint16_t bfloat16_bits = 0b0000000010000000;
+    uint32_t float_bits = 0b00000000100000000000000000000000;
+
+    bfloat16 bf_min = sycl::bit_cast<bfloat16>(bfloat16_bits);
+    float float_min = sycl::bit_cast<float>(float_bits);
```
Comment (keryell) [inline tests/extension/oneapi_bfloat16/bfloat16_api.cpp:74]: Use hexadecimal float notation instead of this hack.

Comment (AmirIpma) [reply tests/extension/oneapi_bfloat16/bfloat16_api.cpp:74]: It was implemented according to the test plan.

**Comment by bader** [comment]
@keryell, ping.

File: tests/extension/oneapi_bfloat16/bfloat16_api.cpp (hunk 2 above)
Comment (keryell) [reply tests/extension/oneapi_bfloat16/bfloat16_api.cpp:74]: Right. Good point.
@gmlueck But perhaps this is an opportunity to improve the test plan to keep the atrocity level a minima? :-)

File: tests/extension/oneapi_bfloat16/bfloat16_api.cpp [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+TEST_CASE("Special values", "[bfloat16]") {
+  SECTION("Check that bfloat16 occupies 16 bits of memory") {
+    REQUIRE(sizeof(bfloat16) == 2);
+  }
+
+  SECTION("Check minimum positive normal value") {
+    uint16_t bfloat16_bits = 0b0000000010000000;
+    uint32_t float_bits = 0b00000000100000000000000000000000;
```
Comment (keryell) [inline tests/extension/oneapi_bfloat16/bfloat16_api.cpp:71]: This is probably https://en.cppreference.com/w/cpp/types/numeric_limits `min` or `denorm_min`.

Comment (bader) [reply tests/extension/oneapi_bfloat16/bfloat16_api.cpp:71]: min

File: tests/extension/oneapi_bfloat16/bfloat16_api.cpp [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+    bfloat16 bf_min = sycl::bit_cast<bfloat16>(bfloat16_bits);
+    float float_min = sycl::bit_cast<float>(float_bits);
+
+    CHECK(bf_min == float_min);
```
Comment (bader) [inline tests/extension/oneapi_bfloat16/bfloat16_api.cpp:76]:
```suggestion
    bfloat16 bf_min = sycl::bit_cast<bfloat16>(0b0000000010000000);

    CHECK(bf_min == std::numeric_limits<float>::min());
```

@keryell, do you mean something like this?

Comment (keryell) [reply tests/extension/oneapi_bfloat16/bfloat16_api.cpp:76]: Yes, that seems quite clearer! :-)

File: tests/extension/oneapi_bfloat16/bfloat16_api.cpp (hunk 2 above)
Comment (gmlueck) [reply tests/extension/oneapi_bfloat16/bfloat16_api.cpp:74]: See #747.

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**